drivers/display/ls0xx.c

```diff
@@ -0,0 +1,493 @@
+/* 3 bufs One for write_cmd/clear_cmd, one for data and one for dummy byte at the end */
+static struct spi_buf buf[3];
+static struct spi_buf_set buf_set = {.buffers   = (struct spi_buf*)&buf,
+                                     .count     = 3};
+
+#if defined(CONFIG_LS0XX_EXTMODE_SERIAL)
+    static uint8_t extcomin_bit = 0x00;
+    #define LS0XX_EXTCOMIN_BIT extcomin_bit
+#else
+    #if !defined(LS0XX_EXTCOMIN_CNTRL)
+        #error "CONFIG_LS0XX_EXTMODE_SERIAL is 'n' or not defined and \
+extcomin-gpios is not present in devicetree"
+    #else
+        #define LS0XX_EXTCOMIN_BIT 0x00
+    #endif
+#endif
+
+struct ls0xx_data {
+    struct device *scs;
+#if defined(LS0XX_DISP_CNTRL)
+    struct device *disp;
+#endif
+#if defined(LS0XX_EXTCOMIN_CNTRL)
+    struct device *extcomin;
+#endif
+    struct device *spi_dev;
+    struct spi_config spi_config;
+};
+
+/**
+ * @brief Toggle VCOM to avoid DC bias
+ *
+ * @details This API will toggle either the EXTCOMIN pin
+ *          or the M1 bit in serial comm to toggle VCOM
+ *
+ * @param driver Pointer to the display driver
+ */
+static inline void toggle_vcom_bit(struct ls0xx_data *driver) {
+#if defined(CONFIG_LS0XX_EXTMODE_SERIAL)
+    extcomin_bit = extcomin_bit ? 0x00 : LS0XX_BIT_VCOM;
+#else
+    gpio_pin_toggle(driver->extcomin, LS0XX_EXTCOMIN_PIN);
+#endif
+}
+
+/**
+ * @brief Display memory contents of the display
+ *
+ * @details Display memory contents, ie actual image data 
+ *          During initialization DISP pin will be set high
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_blanking_off(const struct device *dev) {
+#if defined(LS0XX_DISP_CNTRL)
+    struct ls0xx_data *driver = dev->data;
+    return gpio_pin_set(driver->disp, LS0XX_DISP_PIN, 1);
+#else
+    LOG_ERR("Unsupported");
+    return -ENOTSUP;
+#endif
+}
+
+/**
+ * @brief Display becomes completely white
+ *
+ * @details Displays complete white while but the
+ *          memory contents are retained
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_blanking_on(const struct device *dev) {
+#if defined(LS0XX_DISP_CNTRL)
+    struct ls0xx_data *driver = dev->data;
+    return gpio_pin_set(driver->disp, LS0XX_DISP_PIN, 0);
+#else
+    LOG_ERR("Unsupported");
+    return -ENOTSUP;
+#endif
+}
+
+/**
+ * @brief Clear the internal memory of the display
+ *
+ * @details Clear the display memory and the frame_buf
+ *          setting it to all white
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_clear(const struct device *dev) {
+
+    struct ls0xx_data *driver = dev->data;
+    int err;
+    memset(frame_buf, 0xFF, LS0XX_PANEL_HEIGHT*(((LS0XX_PANEL_WIDTH/LS0XX_PIXELS_PER_BYTE) + 2)));
+
+    uint8_t clear_cmd[2] = {LS0XX_BIT_CLEAR | LS0XX_EXTCOMIN_BIT, 0};
+    toggle_vcom_bit(driver);
+    buf[0].len = sizeof(uint16_t);
+    buf[0].buf = &clear_cmd;
+    buf_set.count = 1;
+
+    gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 1);
```

Comment:
LS0XX_SCS_PIN (serial chip select) should be handled by SPI driver

Reply:
1) This display has an inverted SCS. It is active high. Does the SPI driver support this?
2) I get the following ASSERT when trying to use the SPI driver
```
 /* Validate CS active levels are equivalent */
__ASSERT(spi_context_cs_active_level(ctx) ==
 (ctx->config->cs->gpio_dt_flags & GPIO_ACTIVE_LOW)
```

Reply:
Yes SPI driver API supports it and it is mandatory to describe it, for example https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/shields/waveshare_epaper/waveshare_epaper_gdeh0154a07.overlay#L9